src/widgets/Step.js

```diff
@@ -73,7 +74,7 @@ export const Step = memo(props => {
   useEffect(() => {
     if (status !== 'CURRENT' || !getModalData) return;
     setModalData(getModalData());
-  }, [status, internalStatus]);
+  }, [status, internalStatus, data]);
```

Comment:
Did it take long to find this? One of the arguments I read against prematurely optimising with memoization (might have been from the man Dan?) was that it makes tracing bugs a bit more difficult, so is a compromise against the value of the optimisation

Reply:
This one, no - but I have had others that I've had issues with. To be honest, getting pretty good at spotting when it's a memoization problem just from the behaviour of the interface.. 🤣